srv/wordpress/wp-content/plugins/acore-wp-plugin/src/Hooks/WooCommerce/GuildChange.php

```diff
@@ -0,0 +1,155 @@
+    }
+
+    // 4) Render on checkout
+    public static function get_item_data($cart_data, $cart_item = null) {
+
+        if (isset($cart_item['acore_char_sel'])) {
+            $ACoreSrv = ACoreServices::I();
+            $charRepo = $ACoreSrv->getCharactersRepo();
+
+            $charId = $cart_item['acore_char_sel'];
+            $guildNewName = $cart_item['acore_new_guild_name'];
+
+            $char = $charRepo->findOneByGuid($charId);
+
+            $charName = $char ? $char->getName() : "The character <$charId> does not exist!";
+
+            $custom_items[] = array("name" => 'Character', "value" => $charName);
+            $custom_items[] = array("name" => 'New Guild Name', "value" => $guildNewName);
+        }
+
+        return $custom_items;
+    }
+
+    // 5) ADD DATA TO FINAL ORDER META
+    // This is a piece of code that will add your custom field with order meta.
+    public static function add_order_item_meta($item_id, $values, $cart_item_key) {
+        if ($values['acore_item_sku'] != "guild-rename") {
+            return;
+        }
+
+        if (isset($values['acore_char_sel']) && isset($values["acore_item_sku"])) {
+            \wc_add_order_item_meta($item_id, "acore_char_sel", $values['acore_char_sel']);
+            \wc_add_order_item_meta($item_id, "acore_new_guild_name", $values['acore_new_guild_name']);
+            \wc_add_order_item_meta($item_id, "acore_item_sku", $values['acore_item_sku']);
+        }
+    }
+
+    // 6) check before payment
+    public static function checkout_order_processed($order_id, $posted_data) {
+        $logs = new \WC_Logger();
+
+        $order = new \WC_Order($order_id);
+        $items = $order->get_items();
+
+        $soap = ACoreServices::I()->getServerSoap();
+
+        foreach ($items as $item) {
+            if ($item["acore_item_sku"]) {
+                if ($item["acore_item_sku"] == "guild-rename") {
+                    $res = $soap->serverInfo();
+                    if ($res instanceof \Exception) {
+                        throw new \Exception(__('The server seems to be offline, try again later!', 'woocommerce'));
+                    }
+                    return;
+                }
+            }
+        }
+    }
+
+    // 7)DO THE FINAL ACTION
+    public static function payment_complete($order_id) {
+        $WoWSrv = ACoreServices::I();
+        $logs = new \WC_Logger();
+        try {
+            $order = new \WC_Order($order_id);
+            $items = $order->get_items();
+
+            $soap = $WoWSrv->getGuildSoap();
+            
+            foreach ($items as $item) {
+                if (isset($item["acore_item_sku"])) {
+                    if ($item["acore_item_sku"] == "guild-rename") {
+                        $guildName = $WoWSrv->getGuildNameByLeader($item["acore_char_sel"]);
+
+                        if (empty($guildName))
+                        {
+                            throw new \Exception("The character is not the guild master.");
```

Comment:
can this be checked before? during the char selection?

Reply:
I believe it's checked in both places already, unless I'm mistaken, or I didn't get what you mean specifically